Docs: Adopting Dagger

shykes opened the pull request:
Expand the "Adopting Dagger" page with a framework for successful adoption.

levlaz commented on docs/current_docs/adopting.mdx:
```diff
@@ -4,13 +4,121 @@ slug: /adopting
+Every project is different, and every pipeline is a unique snowflake (usually that's not a good thing). So we can't give you an exact step-by-step guide of how to daggerize your project. But we can share the collective experience of the Dagger community, distilled into a simple framework.
 
-Not sure which part of your pipeline to Daggerize first? Reach out to us on [Discord](https://discord.gg/dagger-io) via our [help forum](https://discord.com/channels/707636530424053791/1030538312508776540). Our team and your fellow Daggernauts will be happy to provide more specific guidance on your use case there! 
+## Phase 1: research
```
Right before this section might be a good place to put a simple diagram showing the three stages.

shykes replied:
Good idea. Let's leave it for a follow-up PR if you don't mind?

levlaz commented on docs/current_docs/adopting.mdx:
```diff
@@ -4,13 +4,121 @@ slug: /adopting
+
+Consider yourself warned: adopting Dagger without joining our Discord is certainly possible, but you probably won't be as productive, and it definitely won't be as fun!
+
+### Look for inspiration
```
We should curate this a bit more, there are many case studies and videos that predate modules, so folks finding those will lead to confusion. 

Perhaps a good place to highlight some of the mature DtW projects? 

If we do link out to more generic indexes, we should provide a pointer letting people know to limit their exploration to things that were published recently.

shykes replied:
Makes sense. I'll also leave that for a follow-up PR (trying to get this one merged before 0.12, so we benefit from the influx of new visitors drawn by the announcement)

levlaz commented on docs/current_docs/adopting.mdx:
```diff
@@ -4,13 +4,121 @@ slug: /adopting
+- You are happily using a monolithic toolchain, such as Gradle, Nix or Bazel, with no exception and no fragmentation within the team
+- Your pipelines are heavily dependent on Windows or Mac runners
+
+In doubt, don't hesitate to ask the community - you've joined the Discord, right?
```
I love the conversational nature of this :)

levlaz commented on docs/current_docs/adopting.mdx:
There is some inconsistent casing for Daggerize vs daggerize, can we pick one?

levlaz commented on docs/current_docs/adopting.mdx:
typo "supported"

```suggestion
- *Check SDK availability*. Dagger supports 3 stable SDKs (Python, Go, Typescript), but it also has several community-contributed SDKs, some of which are actively used in production despite not being officially supported. Take the time to investigate the status of available SDKs, and ask the community for their opinion. Our Discord server is a good place to start - each supported language has a dedicated channel.
```

levlaz commented on docs/current_docs/adopting.mdx:
```suggestion
- If your team prefers (or is forced to) use another tool, whether off-the-shelf like `make`, or custom-built tool or script, it's perfectly fine to wrap your `dagger call` invocations in existing Makefile, configuration files or scripts. Some power users may scratch beneath the surface and call `dagger call` directly; others won't. Both are fine.
```

levlaz commented on docs/current_docs/adopting.mdx:
nit: I think there are too many exclamation points in this section, I'd remove both of the ones here and leave the one at the end of the next paragraph.

levlaz commented on docs/current_docs/adopting.mdx:
nit: mixed metaphors :)

shykes replied:
You're right! Fixed.

levlaz commented on docs/current_docs/adopting.mdx:
I think this is a good place to encourage people to make reusable modules. Ideally abstract public ones that contribute to the daggerverse, but even private ones for internal use should be encouraged at this phase.

shykes replied:
Great idea. Done.

levlaz reviewed (dismissed):
This is excellent overall, I love the conversational nature and the clear direction that it provides to teams of all sizes. 

I left a few comments and typo fixes.

shykes commented:
- Incorporated @levlaz feedback, thank you!
- Added a phase 4: spin out module
- Added a ton of details about CI integration. Was shocked to find that some of this stuff is not written down anywhere. A good reminder that we still have massive gaps.

shykes commented:
Failed check is a flake

jedevc commented on docs/current_docs/adopting.mdx:
:eyes:

shykes replied:
gah

shykes replied:
fixed, thank you

shykes commented:
I'm going to fast track this since it got several thumbs up, and I'd like to take it live before 0.12 is announced

jedevc reviewed (approved):
One tiny typo nit, otherwise LGTM, so expediting :tada:

shykes commented:
Thank you @jedevc ! Fixed the typo you found, rebased, and now merging!
